core-client/transports/src/transports/local.rs

```diff
@@ -97,24 +101,55 @@ where
 	(client, rpc_client)
 }
 
+/// Connects to a `Deref<Target = MetaIoHandler<Metadata>`.
+pub fn connect_with_metadata<TClient, THandler, TMetadata>(
+	handler: THandler,
+	meta: TMetadata,
+) -> (TClient, impl Future<Item = (), Error = RpcError>)
+	where
+		TClient: From<RpcChannel>,
+		THandler: Deref<Target = MetaIoHandler<TMetadata>> + Unpin,
+		TMetadata: Metadata,
+{
+	connect_with_metadata_and_middleware(handler, meta)
+}
+
 /// Connects to a `Deref<Target = MetaIoHandler<Metadata + Default>`.
-pub fn connect<TClient, THandler, TMetadata>(handler: THandler) -> (TClient, impl Future<Item = (), Error = RpcError>)
+pub fn connect_with_middleware<TClient, THandler, TMetadata, TMiddleware>(
+	handler: THandler,
+) -> (TClient, impl Future<Item = (), Error = RpcError>)
 where
 	TClient: From<RpcChannel>,
-	THandler: Deref<Target = MetaIoHandler<TMetadata>>,
+	TMiddleware: Middleware<TMetadata>,
+	THandler: Deref<Target = MetaIoHandler<TMetadata, TMiddleware>> + Unpin,
 	TMetadata: Metadata + Default,
 {
-	connect_with_metadata(handler, Default::default())
+	connect_with_metadata_and_middleware(handler, Default::default())
+}
+
+/// Connects to a `Deref<Target = MetaIoHandler<Metadata + Default>`.
+pub fn connect<TClient, THandler, TMetadata>(
```

Comment:
hrm, this will connect without `Middleware (Noop)` because of `MetaIoHandler<TMetadata>`? It seems unintuitive to me but I understand that it wouldn't be backward compatible otherwise. Perhaps explain this with better documentation and remove it in `v16` later?

Reply:
it will connect with the `NoopMiddleware`

Reply:
Yepp, that was what I meant with `Middleware (Noop)`, but I was mostly referring to:

> Connects to a `Deref<Target = MetaIoHandler<Metadata + Default>`

It's not easy to understand that is using the default associated type `NoopMiddleware` at least not to me.